Coordinate.cpp

```diff
@@ -14,6 +14,14 @@ NodeCoordinate CellCoordinate::toNode(void) const {
     return NodeCoordinate(x * 2 + 1, y * 2 + 1);
 }
 
+bool NodeCoordinate::isCell() {
+    return !((this->x - 1) % 2 || (this->y - 1) % 2);
```

Comment:
ok. Also !(A||B) == (!A && !B)

A iff x is even, B iff y is even

This statement evaluates to true if x and y are odd. Logic is correct.